client/src/pages/messaging/index.js

```diff
@@ -188,6 +188,9 @@ const Chat = () => {
         })
       );
     });
+    socket.on('delivered', () => {
+      setDelivered(true);
+    });
```

Comment:
`delivered` can't be global. It has to be message specific. You can pass some `id` to identify which message is delivered.

Reply:
I understood my mistake. I am working on it

Reply:
`        socket.emit('received');`

I figured this as a way to notify the server that the message has been received. `socketEmit` is where the message is emitted so I also made it to listen to the acknowledgement by the receiver

Reply:
https://github.com/muke1908/chat-e2ee/pull/135/files#r484045037